### PR TITLE
growthAnalyze (choice 1 / simple fix) - only merge one of these

### PR DESCRIPTION
**Pull request by zeddrak**
Implementation of the simplest fix for the growthAnalyze issue.
It will take longer to implement the others.
(Not tested, not carefully examined for knock-on effects)

### CODE REVIEW

**Comment by zeddrak** [comment]
The main point of this version of the fix is to be as simple as possible.
It's still up to the player to hand growthAnalyze a "correct" multiplier (ie, they should hand it infinity if they want to grow a $0 server - and growthAnalyze should handle that).
this fix also brings the actual grow functionality in-line with the behavior growthAnalyze reports.

**Comment by zeddrak** [comment]
BTW: An alternative to using $1 (if that's too punishing) is to use a minimum percentage (like servers are always considered to have at least 0.001% of their max when growing) - this is effectively the same as capping the growth multiplier required to max a zeroed server to a given amount (ie, it'll never take a growth multiplier > 1e5 to cap a zeroed server...) But the simple truth is, even at $1, multiple grows can eventually recover the server given enough time.

The key point of this solution option is to avoid the server's current moneyAvailable as a dependency and thus avoid the irreversible (a+x)**(x*m) issue.

**Comment by zeddrak** [comment]
(PS: The percentage option has potential for exploitation)

File: src/Server/ServerHelpers.ts [hunk 1]
```diff
@@ -74,8 +75,11 @@ export function processSingleServerGrowth(server: Server, threads: number, p: IP
     serverGrowth = 1;
   }
 
+  if (server.moneyAvailable < 1) { //can grow a server reduced to $0
+    server.moneyAvailable = 1;
+  }
   const oldMoneyAvailable = server.moneyAvailable;
-  server.moneyAvailable += 1 * threads; // It can be grown even if it has no money
+  //server.moneyAvailable += 1 * threads; // It can be grown even if it has no money
```
Comment (phyzical) [inline src/Server/ServerHelpers.ts:82]: we might aswell just remove this

**Review by phyzical — COMMENTED** [review]
LGTM,

will approve when test script and gif of it working

**Pull request closed without merging**